lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/fs.rb

```diff
@@ -23,8 +23,14 @@ class Console::CommandDispatcher::Stdapi::Fs
 
   CHECKSUM_ALGORITHMS = %w{ md5 sha1 }
   private_constant :CHECKSUM_ALGORITHMS
-  PATH_EXPAND_REGEX = /\%(\w*)\%/
-  private_constant :PATH_EXPAND_REGEX
+
+  def path_expand_regex
+    if client.platform == 'windows'
+      /\%(\w*)\%/
+    else
+      /\$(([A-Za-z0-9_]+)|\{([A-Za-z0-9_.]+)\})|^~/
+    end
```

Comment:
Again same issue as before here, not sure that we should be using a `else` statement here when there could be non-Unix platforms in the list.

Reply:
```suggestion
    else
      /\$(([A-Za-z0-9_]+)|\{([A-Za-z0-9_]+)\})|^~/
    end
```
Correct me if I am wrong but I don't think periods are allowed in environment variables and the way this is coded right now it would potentially expand out to other characters?

Reply:
> I don't think periods are allowed in environment variables

`.` is not allowed in variable names; and Bash and Dash `export` builtin explicitly forbids it:

```
$ export asdf.zxcv=123.456
bash: export: `asdf.zxcv=123.456': not a valid identifier
```

```
$ export asdf.zxcv=123.456
dash: 1: export: asdf.zxcv: bad variable name
```

> and the way this is coded right now it would potentially expand out to other characters?

`.` is literal when using in `[]` regex character sets:

```
irb(main):001:0> "$asdf.zxcv=123.456".scan(/\$(([A-Za-z0-9_]+)|\{([A-Za-z0-9_.]+)\})|^~/).flatten
=> ["asdf", "asdf", nil]
irb(main):002:0> "${asdf.zxcv}=123.456".scan(/\$(([A-Za-z0-9_]+)|\{([A-Za-z0-9_.]+)\})|^~/).flatten
=> ["{asdf.zxcv}", nil, "asdf.zxcv"]
irb(main):003:0> "${asdf.zxcv=123.456}".scan(/\$(([A-Za-z0-9_]+)|\{([A-Za-z0-9_.]+)\})|^~/).flatten
=> []
irb(main):004:0> "aBc".scan(/[a-z.]+/).flatten
=> ["a", "c"]
irb(main):005:0> "aBc".scan(/([a-z.]+)/).flatten
=> ["a", "c"]
irb(main):006:0> 
```

Reply:
Thanks for clarification @bcoles, appreciate the more detailed examples here :)

Reply:
Fixed in ebd927da44f89faf96a4a256f1f9627c91c698e0